### PR TITLE
[REVIEW] Add `column::release`

### PR DESCRIPTION
**Pull request by jrhemstad**
Closes https://github.com/rapidsai/cudf/issues/3227

### CODE REVIEW

**Comment by jrhemstad** [comment]
rerun tests

**Comment by raydouglass** [comment]
rerun tests

File: cpp/tests/column/column_test.cu [hunk 1]
```diff
@@ -305,3 +307,62 @@ TYPED_TEST(TypedColumnTest, MoveConstructorWithMask) {
...
+
+  cudf::column::contents contents = col.release();
+  EXPECT_EQ(original_data, contents.data->data());
+  EXPECT_EQ(original_mask, contents.null_mask->data());
+  EXPECT_EQ(2u, contents.children.size());
+  EXPECT_EQ(0, col.size());
+  EXPECT_EQ(0, col.null_count());
+  EXPECT_EQ(cudf::data_type{cudf::type_id::EMPTY}, col.type());
+  EXPECT_EQ(0, col.num_children());
+}
```
Comment (harrism) [inline cpp/tests/column/column_test.cu]:
```suggestion
}

```

**Comment by jrhemstad** [comment]
> I wonder if you should comment in the docs that the user must get the type of the column before calling release().

Good point, I think it makes sense to put `data_type` in `contents`.

**Comment by shwina** [comment]
> I'll just add docs to say it's the user's responsibility to query the size/type/null_count before calling release.

Agreed

**Comment by harrism** [comment]
> Nevermind, I think that would be weird to put the `data_type` in contents. If I did that, then why not the size and null count as well? At that point it feels like a violation of "Do One Thing".

I went through that exact thought process before I suggested documenting it. :)

**Pull request merged**